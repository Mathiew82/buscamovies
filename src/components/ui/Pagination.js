import React, { Component } from "react";

export class Pagination extends Component {
  doNotShowFirstPage() {
    return this.props.currentPage === 1 || this.props.currentPage === 2;
  }

  doNotShowPrevPage() {
    return this.props.currentPage === 1;
  }

  doNotShowNextPage() {
    return this.props.currentPage + 1 > this.props.paginationLength;
  }

  doNotShowLastPage() {
    return (
      this.props.currentPage === this.props.paginationLength ||
      this.props.currentPage === this.props.paginationLength - 1
    );
  }

  render() {
    return (
      <nav
        className="pagination is-centered"
        role="navigation"
        aria-label="pagination"
      >
        <ul className="pagination-list">
          <li style={{ display: this.doNotShowFirstPage() && "none" }}>
            <button
              type="button"
              className="pagination-link"
              onClick={this.props.handleClickPage}
            >
              1
            </button>
          </li>
          <li style={{ display: this.doNotShowFirstPage() && "none" }}>
            <span className="pagination-ellipsis">&hellip;</span>
          </li>
          <li style={{ display: this.doNotShowPrevPage() && "none" }}>
            <button
              type="button"
              className="pagination-link"
              onClick={this.props.handleClickPage}
            >
              {this.props.currentPage - 1}
            </button>
          </li>
          <li>
            <button
              type="button"
              className="pagination-link is-current"
              aria-current="page"
              onClick={this.props.handleClickPage}
            >
              {this.props.currentPage}
            </button>
          </li>
          <li style={{ display: this.doNotShowNextPage() && "none" }}>
            <button
              type="button"
              className="pagination-link"
              onClick={this.props.handleClickPage}
            >
              {this.props.currentPage + 1}
            </button>
          </li>
          <li style={{ display: this.doNotShowLastPage() && "none" }}>
            <span className="pagination-ellipsis">&hellip;</span>
          </li>
          <li style={{ display: this.doNotShowLastPage() && "none" }}>
            <button
              type="button"
              className="pagination-link"
              onClick={this.props.handleClickPage}
            >
              {this.props.paginationLength}
            </button>
          </li>
        </ul>
      </nav>
    );
  }
}
