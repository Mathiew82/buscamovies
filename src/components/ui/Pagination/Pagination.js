import React from "react";
import PropTypes from "prop-types";

function Pagination(props) {
  const { currentPage, paginationLength, clickPage } = props;

  const doNotShowFirstPage = () => {
    return currentPage === 1 || currentPage === 2;
  };

  const doNotShowPrevPage = () => {
    return currentPage === 1;
  };

  const doNotShowNextPage = () => {
    return currentPage + 1 > paginationLength;
  };

  const doNotShowLastPage = () => {
    return (
      currentPage === paginationLength || currentPage === paginationLength - 1
    );
  };

  return (
    <nav
      style={{ display: paginationLength < 2 && "none" }}
      className="pagination is-centered"
      role="navigation"
      aria-label="pagination"
    >
      <ul data-testid="pagination-list" className="pagination-list">
        <li style={{ display: doNotShowFirstPage() && "none" }}>
          <button type="button" className="pagination-link" onClick={clickPage}>
            1
          </button>
        </li>
        <li style={{ display: doNotShowFirstPage() && "none" }}>
          <span className="pagination-ellipsis">&hellip;</span>
        </li>
        <li style={{ display: doNotShowPrevPage() && "none" }}>
          <button type="button" className="pagination-link" onClick={clickPage}>
            {currentPage - 1}
          </button>
        </li>
        <li>
          <button
            type="button"
            className="pagination-link is-current"
            aria-current="page"
            onClick={clickPage}
          >
            {currentPage}
          </button>
        </li>
        <li style={{ display: doNotShowNextPage() && "none" }}>
          <button type="button" className="pagination-link" onClick={clickPage}>
            {currentPage + 1}
          </button>
        </li>
        <li style={{ display: doNotShowLastPage() && "none" }}>
          <span className="pagination-ellipsis">&hellip;</span>
        </li>
        <li style={{ display: doNotShowLastPage() && "none" }}>
          <button type="button" className="pagination-link" onClick={clickPage}>
            {paginationLength}
          </button>
        </li>
      </ul>
    </nav>
  );
}

Pagination.propTypes = {
  currentPage: PropTypes.number,
  paginationLength: PropTypes.number,
  clickPage: PropTypes.func,
};

export default Pagination;
