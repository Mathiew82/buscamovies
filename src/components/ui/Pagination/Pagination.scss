.pagination {
  margin: 2rem 0;
  .pagination-link {
    background-color: var(--bg-color) !important;
    border-color: rgb(104, 86, 128) !important;
    color: white;
    &.is-current {
      background-color: white !important;
      border-color: rgba(0, 0, 0, 0.4) !important;
      color: var(--bg-color) !important;
      cursor: default;
      transition: background-color 0.2s;
    }
    @media (min-width: 769px) {
      &:hover {
        background-color: var(--secondary-color) !important;
        border-color: var(--secondary-color) !important;
        color: var(--bg-color) !important;
        cursor: pointer;
      }
    }
  }
}
