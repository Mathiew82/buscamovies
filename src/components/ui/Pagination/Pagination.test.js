import React from "react";
import { render } from "@testing-library/react";
import Pagination from "./Pagination";

describe("Pagination Component Tests", () => {
  test("Component should render correctly", () => {
    const handleClickPage = () => {};
    const currentPage = 1;
    const paginationLength = 200;

    const { asFragment, unmount } = render(
      <Pagination
        currentPage={currentPage}
        paginationLength={paginationLength}
        handleClickPage={handleClickPage}
      />
    );

    const firstRender = asFragment();
    expect(firstRender).toMatchSnapshot(asFragment());
    unmount();
  });
});
