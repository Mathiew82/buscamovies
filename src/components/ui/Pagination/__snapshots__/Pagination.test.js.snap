// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Component Tests Component should render correctly 1`] = `Object {}`;
